__tests__/reviews.test.js

```diff
@@ -222,11 +219,111 @@ describe("GET /api/reviews", () => {
+  });
+  test("status 200: reviews are sorted by comment count", () => {
+    return request(app)
+      .get("/api/reviews?sort_by=comment_count")
+      .expect(200)
+      .then(({ body }) => {
+        const { reviews } = body;
+        expect(reviews).toBeSortedBy("comment_count", {
+          descending: true,
+        });
+      });
+  });
+  test("status 200: reviews are sorted ASC by default sort_by", () => {
+    return request(app)
+      .get("/api/reviews?order=asc")
+      .expect(200)
+      .then(({ body }) => {
+        const { reviews } = body;
+        expect(reviews).toBeSorted({ ascending: true });
+        expect(reviews).toBeSortedBy("created_at", {
+          coerce: true,
+        });
+      });
+  });
+  test("status 200: reviews are sorted ASC by votes", () => {
+    return request(app)
+      .get("/api/reviews?sort_by=votes&order=asc")
+      .expect(200)
+      .then(({ body }) => {
+        const { reviews } = body;
+        expect(reviews).toBeSortedBy("votes");
+        expect(reviews).toBeSorted({ ascending: true });
+      });
+  });
+  test("status 200: reviews are filtered by category", () => {
+    return request(app)
+      .get("/api/reviews?category=euro%20game")
+      .expect(200)
+      .then(({ body }) => {
+        const { reviews } = body;
+        expect(reviews).toBeInstanceOf(Array);
+        expect(reviews).toHaveLength(1);
+        expect(reviews[0].category).toBe('euro game')
+      });
+  });
+  test("status 200: reviews are filtered by category, and ordered in asc order by votes", () => {
+    return request(app)
+      .get("/api/reviews?category=social%20deduction&order=asc&sort_by=votes")
+      .expect(200)
+      .then(({ body }) => {
+        const { reviews } = body;
+        expect(reviews).toBeInstanceOf(Array);
+        expect(reviews).toHaveLength(11);
+        expect(reviews).toBeSortedBy("votes");
+        expect(reviews).toBeSorted({ ascending: true });
+        reviews.forEach((review) => {
+          expect(review).toEqual(
+            expect.objectContaining({
+              category:'social deduction'
+            })
+          )
+        })
+      });
+  });
+
+  test("status 400: responds with 'Bad Request' when invalid sort_by is passed", () => {
+    return request(app)
+      .get("/api/reviews?sort_by=number")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+  test("status 400: responds with 'Bad Request' when invalid order_by is passed", () => {
+    return request(app)
+      .get("/api/reviews?order=number")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+  test("status 404: responds with 'Not Found' when entered a non-existing category", () => {
+    return request(app)
+      .get("/api/reviews?category=number")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Not Found");
+      });
+  });
+});

```

Comment:
Really lovely work here Kristina - very comprehensive - I'd just add a test for a valid category query where there are no reviews yet.